### PR TITLE
#6117 decode token data

### PR DESCRIPTION
**Pull request by mattcowen**

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/TextFieldElementDriver.cs [hunk 1]
```diff
@@ -79,7 +80,7 @@ public class TextFieldElementDriver : FormsElementDriver<TextField>{
         protected override void OnDisplaying(TextField element, ElementDisplayContext context) {
             context.ElementShape.ProcessedName = _tokenizer.Replace(element.Name, context.GetTokenData());
             context.ElementShape.ProcessedLabel = _tokenizer.Replace(element.Label, context.GetTokenData());
-            context.ElementShape.ProcessedValue = _tokenizer.Replace(element.RuntimeValue, context.GetTokenData());
+            context.ElementShape.ProcessedValue = HttpUtility.HtmlDecode(_tokenizer.Replace(element.RuntimeValue, context.GetTokenData()));
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/TextFieldElementDriver.cs]: The actual fix is to not encode it in the first place, by using the correct `Replace` overload. But please check with @sfmskywalker if this is the correct behavior. Or anyone who could confirm.

Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.DynamicForms/Drivers/TextFieldElementDriver.cs]: @sebastienros You were right regarding the security issue. If you put at least two fields on a form, one of which being required and the other one providing a token, that token gets evaluated and displayed (due to the model validation error the form gets displayed and submitted values are shown). Fixing this.

**Comment by sfmskywalker** [comment]
Fixed in db564d1670854459b176459b432833202f98e157.

**Comment by sebastienros** [comment]
Why not fix it on 1.9.x?

**Comment by sfmskywalker** [comment]
No reason. It should have been done on 1.9.x. I'll take care of it.

**Pull request closed without merging**